### ISSUE TITLE
Checking whether a `Literal["..."]` is in a `tuple[T, ...]` is marked as unnecessary

### ISSUE THREAD

**Issue by thesadru** [microsoft/pyright#3739]
```py
import typing

T = typing.TypeVar("T")


def func(x: tuple[T, ...]) -> tuple[T, ...]:
    # Expression will always evaluate to False since the types "Literal['literal']" and "T@func" have no overlap (reportUnnecessaryContains)
    if "literal" in x:
        return ()

    return x
```

Introduced in 25f01e93c9f2f37c27024a3f6342548fc1010ab8

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.264, which I just published. It will also be included in a future release of pylance.
